app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/PollingNewMessagesWithWorkItemJob.scala

```diff
@@ -155,7 +155,7 @@ class PollingNewMessagesWithWorkItemJob @Inject()
       case Nil => Future(false)
       case movements =>
         if (!appConfig.pushNotificationsEnabled) {
-          movements match {
+          movements.toList match {
             case Nil => successful(false)
             case _ :: _ => successful(true)
           }
```

Comment:
Rather than doing a toList, this could be more cost-effective if you used a ternary if on movements.isEmpty.

Further, what on earth is happening on 155 that is different than 159 for you to check if something is Nil twice?